### DISCUSSION TITLE
appdata temp files

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5534] [Q&A]
Can the files in this folder be deleted? Some data back a couple of years. Therea are also a bunch of folders that begin with  jetty-0.0.0.0-8080-webadmin.war that contains files in a webapp folder that are fairly old.

**Comment by jonbartels** [comment]
Old forum post - no answer though. https://forums.mirthproject.io/forum/mirth-connect/support/16239-appdata-temp

I know these files can be deleted between restarts. I have containers which do not preserve this directory and they run fine if its nuked between starts. 

Within that directory are precompiled JSP and webapp files. 

I did a quick experiment.
* I had a running MC instance
* I ran `ls -lathr` on `appdata/temp` and I see that the `jetty` directories match with my container starts
* I restarted my instance
* I ran `ls -lathr` again and I see a new jetty directory

This tells me that these directories are created on startup (or maybe first access).

I continued the experiment:
* I ran `rm -rf appdata/tmp/jetty*` 
* I could no longer access the MC web dashboard
* I _can_ still access MC using the MC Launcher

This experiment suggests that you can delete all but the newest Jetty temp folder. Please do not try this in PROD only on my experiment, you could probably try it in TEST though.

Perhaps an improvement to MC would be to use a constant directory for Jetty temp or for MC to clean up after itself if it finds old paths.

**Reply by jonbartels** [reply]
Jetty seems to use this order - https://stackoverflow.com/questions/19232182/jetty-how-to-change-startup-temp-directory/19232771#19232771

I see in MC that the `dir.tempdata` property defines the path, then MC overwrites `java.io.tmpdir`. 

I'm reading https://www.eclipse.org/jetty/documentation/jetty-9/index.html#ref-temporary-directories but my coffee break is almost over. I see how Jetty can be told to use a different temp dir but I do not see how to tell MC to use the same sub-directory under temp for each Jetty WAR.

**Reply by jonbartels** [reply]
`setPersistTempDirectory` might be the magic beans here for Mirth if you wanted to file an issue to make MC clean up after itself or re-use the same directory.  It could also be something like Jetty will clean up after itself on a clean shutdown but MC doesnt call that.

I'm on a tangent. You can safely delete the jetty folders in appdata/temp on a cold system. Its TEMP. The only problem you might have is if you delete temp files under a running MC instance.
